numba/cpython/listobj.py

```diff
@@ -921,41 +938,19 @@ def list_extend(lst, iterable):
 
     return context.compile_internal(builder, list_extend, sig, args)
 
-@lower_builtin("list.index", types.List, types.Any)
-def list_index(context, builder, sig, args):
 
-    def list_index_impl(lst, value):
-        for i in range(len(lst)):
-            if lst[i] == value:
-                return i
-        # XXX references are leaked when raising
-        raise ValueError("value not in list")
+intp_max = types.intp.maxval
 
-    return context.compile_internal(builder, list_index_impl, sig, args)
 
-@lower_builtin("list.index", types.List, types.Any,
-           types.Integer)
-def list_index(context, builder, sig, args):
+@overload_method(types.List, "index")
+def list_index(lst, value, start=0, stop=intp_max):
 
-    def list_index_impl(lst, value, start):
-        n = len(lst)
-        if start < 0:
-            start += n
-            if start < 0:
-                start = 0
-        for i in range(start, len(lst)):
-            if lst[i] == value:
-                return i
-        # XXX references are leaked when raising
-        raise ValueError("value not in list")
+    if not isinstance(start, (int, types.Integer, types.Omitted)):
+        raise TypeError(f'arg "start" must be an Integer. Got {start}')
+    if not isinstance(stop, (int, types.Integer, types.Omitted)):
+        raise TypeError(f'arg "end" must be an Integer. Got {end}')
```

Comment:
Just noticed that this refers to `end`, which is undefined. That it's not been caught by testing suggests that it is untested. Perhaps add a test to hit these two branches?